packages/react/src/use-render/useRender.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import type { ComponentRenderFn } from '../utils/types';
+import { useComponentRenderer } from '../utils/useComponentRenderer';
+import { defaultRenderFunctions } from '../utils/defaultRenderFunctions';
+import { CustomStyleHookMapping } from '../utils/getStyleHookProps';
+
+/**
+ * Returns a function that renders a Base UI component.
+ */
+function useRender<State extends Record<string, unknown>, RenderedElementType extends Element>(
+  settings: useRender.Settings<State, RenderedElementType>,
+) {
+  const { render, props, state = {} } = settings;
+  const { ref, ...extraProps } = props ?? {};
+
+  return useComponentRenderer({
+    render,
+    state: state as State,
+    ref: ref as React.Ref<RenderedElementType>,
+    extraProps,
+    customStyleHookMapping: Object.keys(state).reduce((acc, key) => {
+      acc[key as keyof State] = () => null;
+      return acc;
+    }, {} as CustomStyleHookMapping<State>),
+  });
```

Comment:
This calculation and `state` should be memo'd

Reply:
I've added the calculation in a useMemo, I will add a note for the state to be memoized in the docs, is that what you had in mind?